cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -1390,6 +1591,58 @@ void serialize_to_hnswlib(raft::resources const& handle,
 void serialize_to_hnswlib(raft::resources const& handle,
                           const std::string& filename,
                           const cuvs::neighbors::cagra::index<float, uint32_t>& index);
+/**
+ * Write the CAGRA built index as a base layer HNSW index to an output stream
+ *
+ * Experimental, both the API and the serialization format are subject to change.
+ *
+ * @code{.cpp}
+ * #include <raft/core/resources.hpp>
+ * #include <cuvs/neighbors/cagra_serialize.hpp>
+ *
+ * raft::resources handle;
+ *
+ * // create an output stream
+ * std::ostream os(std::cout.rdbuf());
+ * // create an index with `auto index = raft::cagra::build(...);`
+ * cuvs::neighbors::cagra::serialize_to_hnswlib(handle, os, index);
+ * @endcode
+ *
+ * @param[in] handle the raft handle
+ * @param[in] os output stream
+ * @param[in] index CAGRA index
+ *
+ */
+void serialize_to_hnswlib(raft::resources const& handle,
+                          std::ostream& os,
+                          const cuvs::neighbors::cagra::index<half, uint32_t>& index);
```

Comment:
Just curious, does HNSW support half data type or we convert it internally to fp32? If the latter, perhaps it makes sense to document this.